rest-service/manager_rest/resources_v2.py

```diff
@@ -92,6 +93,204 @@ def _create_filter_params_list_description(parameters, list_type):
+
+    @swagger.operation(
+        responseClass=responses_v2.Snapshot,
+        nickname='createSnapshot',
+        notes='Create a new snapshot of the manager.',
+        consumes=[
+            "application/json"
+        ]
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Execution)
+    def put(self, snapshot_id):
+        verify_json_content_type()
+        request_json = request.json
+        verify_parameter_in_request_body('include_metrics', request_json)
+        verify_parameter_in_request_body('include_credentials', request_json)
+        include_metrics = verify_and_convert_bool(
+            'include_metrics',
+            request_json['include_metrics']
+        )
+        include_credentials = verify_and_convert_bool(
+            'include_credentials',
+            request_json['include_credentials']
+        )
+
+        execution = get_blueprints_manager().create_snapshot(
+            snapshot_id,
+            include_metrics,
+            include_credentials
+        )
+        return execution, 201
+
+    @swagger.operation(
+        responseClass=responses_v2.Snapshot,
+        nickname='deleteSnapshot',
+        notes='Delete existing snapshot.'
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Snapshot)
+    def delete(self, snapshot_id):
+        snapshot = get_blueprints_manager().delete_snapshot(snapshot_id)
+        path = _get_snapshot_path(snapshot_id)
+        shutil.rmtree(path, ignore_errors=True)
+        return snapshot, 200
+
+    @exceptions_handled
+    def patch(self, snapshot_id):
+        """
+        Update snapshot status by id
+        """
+        verify_json_content_type()
+        request_json = request.json
+        verify_parameter_in_request_body('status', request_json)
+
+        get_blueprints_manager().update_snapshot_status(
+            snapshot_id,
+            request_json['status'],
+            request_json.get('error', ''))
+
+
+class SnapshotsIdArchive(SecuredResource):
+
+    @swagger.operation(
+        responseClass=responses_v2.Snapshot,
+        nickname='uploadSnapshot',
+        notes='Submitted snapshot should be an archive.'
+              'Archive format has to be zip.'
+              ' Snapshot archive may be submitted via either URL or by '
+              'direct upload.',
+        parameters=[{
+            'name': 'snapshot_archive_url',
+            'description': 'url of a snapshot archive file',
+            'required': False,
+            'allowMultiple': False,
+            'dataType': 'string',
+            'paramType': 'query'
+        }, {
+            'name': 'body',
+            'description': 'Binary form of the zip',
+            'required': True,
+            'allowMultiple': False,
+            'dataType': 'binary',
+            'paramType': 'body'}],
+        consumes=[
+            "application/octet-stream"
+        ]
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Snapshot)
+    def put(self, snapshot_id):
```

Comment:
This API is not consistent with other APIs we have.
/archive is only used for downloading the resource (for instance, /blueprints/<id>/archive).
For being consistent, upload snapshot from URL should be a flag in request body as implemented for uploading a blueprint from URL.